exporters/otlp/CMakeLists.txt

```diff
@@ -226,6 +226,7 @@ if(BUILD_TESTING)
   endif()
 
   if(WITH_OTLP_GRPC)
+    add_compile_options(-Wno-suggest-override)
```

Comment:
As far as I remember, if OTELCPP_MAINTAINER_MODE is enabled, -Werror is set. So, with one of the 3rd party libraries, "override" warning just blocks the compilation due to this reason.

Reply:
The maintainer mode is used in CI and builds cleanly.

Changing includes and adding forward declarations should not cause warnings in virtual methods, so this needs to be investigated instead of silenced.

Please fix unbalanced thrift_include_prefix / suffix, it interferes with warnings also.

Reply:
Also removed this compile option, if there is no side effect on compile error.